pkg/p2p/peer/connector.go

```diff
@@ -0,0 +1,161 @@
+// This Source Code Form is subject to the terms of the MIT License.
+// If a copy of the MIT License was not distributed with this
+// file, you can obtain one at https://opensource.org/licenses/MIT.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+package peer
+
+import (
+	"bytes"
+	"context"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/dusk-network/dusk-blockchain/pkg/config"
+	"github.com/dusk-network/dusk-blockchain/pkg/p2p/wire/protocol"
+	"github.com/dusk-network/dusk-blockchain/pkg/util/nativeutils/eventbus"
+	log "github.com/sirupsen/logrus"
+)
+
+// Connector is responsible for accepting incoming connection requests, and
+// establishing outward connections with desired peers.
+type Connector struct {
+	eventBus      eventbus.Broker
+	gossip        *protocol.Gossip
+	readerFactory *ReaderFactory
+
+	l net.Listener
+
+	lock     sync.RWMutex
+	registry map[string]struct{}
+
+	services protocol.ServiceFlag
+}
+
+// NewConnector creates a new peer connector, and spawns a goroutine that will
+// accept incoming connection requests on the current address, with the given port.
+func NewConnector(eb eventbus.Broker, gossip *protocol.Gossip, port string,
+	processor *MessageProcessor, services protocol.ServiceFlag) *Connector {
+	addrPort := ":" + port
+
+	listener, err := net.Listen("tcp", addrPort)
+	if err != nil {
+		log.WithField("process", "peer listener").
+			WithError(err).
+			Panic("could not establish a listener")
+	}
+
+	c := &Connector{
+		eventBus:      eb,
+		gossip:        gossip,
+		readerFactory: NewReaderFactory(processor),
+		l:             listener,
+		registry:      make(map[string]struct{}),
+		services:      services,
+	}
+
+	go func(c *Connector) {
+		for {
+			conn, err := c.l.Accept()
+			if err != nil {
+				log.WithField("process", "connection manager").
+					WithError(err).
```

Comment:
would you agree that for debugging proposes would make sense to also add more info such as addrPort ?